app/assets/js/components/Work/Work.tsx

```diff
@@ -15,21 +15,35 @@ const Work = ({ work }: { work: WorkType }) => {
   /**
    * Get the DC API super user token from the API every 5 minutes.
    */
-  const { data: dataDcApiToken, error: errorDcApiToken } = useQuery(
-    GET_DC_API_TOKEN,
-    { pollInterval: 300000 },
-  );
+  const {
+    data: dataDcApiToken,
+    error: errorDcApiToken,
+    refetch: refetchDcApiToken,
+  } = useQuery(GET_DC_API_TOKEN);
 
-  const dcApiToken = dataDcApiToken?.dcApiToken?.token;
+  const { token, expires } = dataDcApiToken?.dcApiToken || {};
 
-  useEffect(
-    () =>
+  useEffect(() => {
+    if (token) {
       workDispatch({
         type: "updateDcApiToken",
-        dcApiToken,
-      }),
-    [dcApiToken],
-  );
+        dcApiToken: token,
+      });
+    }
+
+    const handleTokenUpdate = setInterval(() => {
+      const timePadding = 60000; // 1 minute
+      const currentDate = new Date();
+      const expiresDate = new Date(expires);
+      const expireDifference = expiresDate.getTime() - currentDate.getTime();
+      console.log(currentDate, expiresDate, expireDifference);
+      if (expireDifference < timePadding) {
+        refetchDcApiToken();
+      }
+    }, 100);
```

Comment:
@mathewjordan Does this need to be such a short interval?  Looks like the component re-renders are reasonable 👍, but this interval is firing continuously...not sure how much of a performance hit, if any it is tho...

Reply:
We need some sort of interval here. It could be a little looser, but all that is happening is the conditional on line 40 which shouldn't be too intensive.

Reply:
Yea, resource intensive probably not that big a hit but it is something just running non-stop which fires up my OCD 😀.  What if we either bumped that interval value a touch (`300`?), _or_ what about something more targeted like this?

```
useEffect(() => {
    console.log("useEffect()");

    let handleTokenUpdate: NodeJS.Timeout | null = null;

    if (token) {
      workDispatch({
        type: "updateDcApiToken",
        dcApiToken: token,
      });

      const timePadding = 60000; // 1 minute
      const currentDate = new Date();
      const expiresDate = new Date(expires);
      const expireDifference = expiresDate.getTime() - currentDate.getTime();

      if (expireDifference > timePadding) {
        console.log(
          "expireDifference > timePadding",
          expireDifference,
          timePadding,
        );
        handleTokenUpdate = setTimeout(() => {
          console.log("refetchDcApiToken()", expireDifference - timePadding);
          refetchDcApiToken();
        }, expireDifference - timePadding);
      }
    }

    return () => {
      if (handleTokenUpdate) {
        clearTimeout(handleTokenUpdate);
      }
    };
  }, [token, expires, refetchDcApiToken, workDispatch]);
```

Reply:
Convenience `console.logs` left in above...